src/main/java/org/datadog/jenkins/plugins/datadog/traces/DatadogTraceBuildLogic.java

```diff
@@ -142,6 +143,9 @@ public void finishBuildTrace(final BuildData buildData, final Run<?,?> run) {
         buildSpan.setTag(CITags.GIT_REPOSITORY_URL, gitUrl);
 
         final String gitCommit = buildData.getGitCommit("").isEmpty() ? updatedBuildData.getGitCommit("") : buildData.getGitCommit("");
+        if(!isValidCommit(gitCommit)) {
+            logger.warning("Couldn't find a valid commit for pipelineID '"+buildData.getBuildTag("")+"'. GIT_COMMIT environment variable was not found or has invalid SHA1 string: " + gitCommit);
+        }
         buildSpan.setTag(CITags.GIT_COMMIT__SHA, gitCommit); //Maintain retrocompatibility
         buildSpan.setTag(CITags.GIT_COMMIT_SHA, gitCommit);
```

Comment:
Should this still happen if gitCommit is not valid?

Reply:
I'd say yes because we need git information to correlate traces for pipeline/stages/jobs spans (atm). We're showing a warning in the logs, though. 

If the commit is not "valid", that means the user modified it explicitly... so, we may want to show what the user set (most probably a short version of the commit).